rust/lsp-lib/src/lsp_plugin.rs

```diff
@@ -372,6 +365,102 @@ impl Plugin for LspPlugin {
     }
 
     fn config_changed(&mut self, _view: &mut View<Self::Cache>, _changes: &ConfigTable) {}
+
+    fn get_hover(
+        &mut self,
+        view: &mut View<Self::Cache>,
+        request_id: usize,
+        position: CorePosition,
+    ) {
+        let rev = view.rev;
+        let view_info = self.view_info.get_mut(&view.get_id());
+        let view_id = view.get_id();
+
+        let view_info = if let Some(view_info) = view_info {
+            view_info
+        } else {
+            return;
+        };
+
+        // This won't fail since we definitely have a client for the given
+        // client identifier
+        let ls_client = self
+            .language_server_clients
+            .get(&view_info.ls_identifier)
+            .unwrap();
+
+        let position_ls = lsp_position_from_core_position(view, position);
```

Comment:
Didn't we decide to do this conversion in core? We shouldn't need to have conversion logic in both core and in the plugin lib. (see my top level comment)